learing_journal/views/default.py

```diff
@@ -1,76 +1,121 @@
+            headers = remember(request, username)
+            return HTTPFound(location=request.route_url('home'), headers=headers)
+    return {}
+
+
+@view_config(route_name='logout')
+def logout(request):
+    headers = forget(request)
+    return HTTPFound(request.route_url('home'), headers=headers)
+
+
+# TODO: test the routes
+# TODO: add if request.method == 'DELETE':
+@view_config(route_name='home', renderer='templates/home.jinja2', permission='view')
+def home(request):
     try:
-        query = request.dbsession.query(MyModel)
-        all_entries = query.all()
+        query = request.dbsession.query(Journal)
+        all_entries = query.order_by(desc(Journal.date)).all()
     except DBAPIError:
         return Response(db_err_msg, content_type='text/plain', status=500)
-    return {'entries': all_entries, 'poject': 'learning_journal'}
+    return {'entries': all_entries}
 
 
-@view_config(route_name='create', renderer='templates/new-entry.jinja2')
+@view_config(route_name='create',
+             renderer='templates/new-entry.jinja2',
+             permission='secret')
 def create(request):
-    return {'poject': 'learning_journal'}
+    '''handles the POST request to create a new entry'''
+    title = body = error = ''
 
+    if request.method == 'POST':
+        title = request.POST['title']
+        body = request.POST['body']
+        print('\nTitle: {}'.format(title))
+        print('Body: {}\n'.format(body))
 
-@view_config(route_name='update', renderer='templates/edit-entry.jinja2')
-@view_config(route_name='detail', renderer='templates/single-entry.jinja2')
+        if title != '' and body != '':
+            date = datetime.datetime.now()
+            date_last_updated = datetime.datetime.now()
+            new = Journal(title=title, body=body, date=date,
+                          date_last_updated=date_last_updated)
+            request.dbsession.add(new)
+            return HTTPFound(location=request.route_url('home'))
+        else:
+            error = "You are missing the Title or the Body"
+            return {'title': title, 'body': body, 'error': error}
+    return {'title': title, 'body': body, 'error': error}
+
+
+@view_config(route_name='update', renderer='templates/edit-entry.jinja2', permission='view')
+@view_config(route_name='detail', renderer='templates/single-entry.jinja2', permission='secret')
 def detail(request):
+    '''handles the GET and POST method for edit-entry and single-entry'''
+
     try:
-        query = request.dbsession.query(MyModel)
+        query = request.dbsession.query(Journal)
         single_entry = query.filter_by(id=request.matchdict['id']).first()
+        if request.method == 'POST':
+            single_entry.title = request.POST['title']
+            single_entry.body = request.POST['body']
+            return HTTPFound(location=request.route_url('home'))
     except DBAPIError:
         return Response(db_err_msg, content_type='text/plain', status=500)
```

Comment:
I'd get rid of this if I were you, and just let the DBAPIError pass unhandled.  The framework will take care of catching it and convert it to a 500 response for you.